src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -1121,7 +1121,7 @@ export class NotebookModel extends Disposable implements INotebookModel {
 			// Switching from Kusto to another kernel should set the currentKernelAlias to undefined
 			if (this._selectedKernelDisplayName !== this._currentKernelAlias && this._selectedKernelDisplayName) {
 				this._currentKernelAlias = undefined;
-			} else {
+			} else if (this._currentKernelAlias) {
```

Comment:
This check is added to avoid setting 'undefined' alias, as I noticed that happening while debugging.